### DISCUSSION TITLE
Page files with no `.page.*` suffix

### DISCUSSION THREAD

**Discussion by davenquinn** [vikejs/vike#863] [Discussion]
I'm trying to make this plugin work for a website in which pages will be constructed from Markdown notes linked together with Wikilink syntax (aligned with [Obsidian](https://obsidian.md)). These tools generally don't provide control over the file suffix, so I am restricted to `<page-slug>.md`.

On Next.js, having these files in my `pages/` folder is adequate to serve them with a default set of MDX templates and and styles. However, I am unsure how to make this structure work easily with this plugin. The options seem either to be a catch-all route (with some logic to replicate Vite's MDX rendering) or by adjusting the page resolution system to accept Markdown files. Any suggestions on how to make this structure work.

**Comment by brillout** [comment]
The [V1 design](https://github.com/brillout/vite-plugin-ssr/issues/578#issuecomment-1528829634) removes the `.page.` suffixes. Does that solve your problem?

**Comment by davenquinn** [comment]
That probably does. I didn't realize that was part of the redesign, or that the new design was in a usable state. I'll try it out and report back.

**Comment by davenquinn** [comment]
Hi again. I've checked this out, and I am not sure it quite fits the bill. As far as I can tell, the suffix is no longer necessary, but all pages must be named `+Page.<ext>` and exist in their own directory? This is slightly more workable, maybe, but will require a lot of ongoing clutter to make WikiLinks work properly.

Is there planned to be any way to serve a tree of pages from simple files (see example below)? It seems that a single routing strategy/config could be defined for the entire tree in a single or few `+config.ts` files.

![image](https://github.com/brillout/vite-plugin-ssr/assets/1537910/6b858972-0baa-499a-8f0e-c8ef91cd22e2)

The broad goal is to have the `/pages` directory strip out as much complexity as possible to enable ~friendly editing of Markdown by contributors who don't understand the structures needed by the website building strategy.

**Reply by brillout** [reply]
This is indeed a concern for which I didn't find any satisfactory solution yet.

[Single Route File](https://github.com/brillout/vite-plugin-ssr/issues/341) offers a solution but it requires adding each `pages/[a-zA-Z].mdx` to `pages/+config.js`.

If [sponsoring](https://github.com/sponsors/brillout) is an option, then I'd be up to increase the priority of this.

**Reply by davenquinn** [reply]
I do see that your approach seems to broadly parallel the new design of NextJS 13, which I wasn't aware of. So I suspect (based on this convergent/parallel evolution) that the approach taken is broadly correct and will strive to adapt my needs to it.

Perhaps the best approach is to write a catch-all route that finds and renders the appropriate content file from an adjacent directory tree? All pages will have essentially the same design and wrapper.

**Reply by brillout** [reply]
You can try that but I think the best would be to have a built-in solution.

**Reply by brillout** [reply]
How about this:

```js
// /pages/+config.js

export default {
  flatPages: {
    defaultExportDefines: 'Page'
  }
}
```

**Reply by davenquinn** [reply]
I suspect that would work quite well. Perhaps there could be a regex pattern to define which files are interpreted as pages within the tree?

**Reply by brillout** [reply]
I don't think it's necessary, especially given that each page requires to create its own directory otherwise.

**Reply by davenquinn** [reply]
Ah that's true.

**Reply by davenquinn** [reply]
...although, if files that weren't intended to be pages (e.g., stylesheets, images, etc.) were in the same directory tree, this would presumably try to build pages for all of them. It'd be ideal to be able to specify that, e.g, only `**/*.mdx?` get dragged in.

**Reply by brillout** [reply]
Yea makes sense.
